### ISSUE TITLE
Monthly Report- Whats left

### ISSUE THREAD

**Issue by hugo-e-pigott** [Vanuatu-National-Statistics-Office/vnso-RAP-tradeStats-materials#94]
Script 1- Processing 

- Will share the CSV that is immediately downloaded from the Customs system for April. Will delete the identifiable data. Will have to write code to potentially delete these columns and rows. Potentially do this at the end and produce two csv's, one that is anonymised and another with individual information

- Converting SITC to character (line 57), SITC Code 8431.2 isn't in convertible format. Anna checked this in HS Code tables that have corresponding SITC codes, and in fact SITC code should be 0744.92. Have gone back into raw data and changed. Now running no issues

- Explore extent of missing data (line 84), checking for large amounts of missing data (>0.1). Can we also look at missing data if only a few cells, maybe create data-frames for each column with missing data 9similar to PRF data-frame)?

- Extract 3 and 1 digit SITC Codes (line 100 & 101), two null or NA values in SITC column skipped (HS Codes- 71189000 & 99050000). Checked HS Tables, document states HS Code 71189000 corresponding SITC is null (will have to confirm with Customs), but HS Code 99050000 corresponding SITC is 0990.15- have gone back into raw data and changed. Now only one running issue. 

- NAs present in any of the above columns (line 104), code View(tradeStatsCommodities[is.na(tradeStatsCommodities$column_name), ]) may not be working

- Been through checkingMergingColumnsForClassificationTables() function, would it be possible to quickly explain how this works? Also not sure of the format of the data-frame produced "missingValuesHSCode_2" could explain also? 

- Merging SITC with cleaned data (line 158). Warning message 1 code in SITC cleaned data that not in classifications. Again not sure of format of data-frame "missingValuesSITC_1" (this to be similar to other merges)

- Merging Import with cleaned data (line 171). Warning message "185 codes in the "CO" column of the Import country classification table are not in the "CO" column of the trades statistics table"- very possible. 

-  Merging export with cleaned data (line 184). Warning message "194 codes in the "CE.CD" column of the Export country classification table are not in the "CE.CD" column of the trades statistics table"- very possible 

- Merging principle commodities with cleaned data (line 198). Warning message "378 codes in the "HS.Code" column of the trades statistics table are not in the "HS.Code" of the Principle Commodities classification table". Need to invest age further and add missing information to classifications table. Also how is this possible when merge previously change in observations is only about 5? Another warning message "3759 codes in the "HS.Code" column of the Principle Commodities classification table are not in the "HS.Code" column of the trades statistics table"- very possible 

- Merging BEC values with cleaned data (line 211). Warning message "611 codes in the "HS.Code_6" column of the trades statistics table are not in the "HS.Code_6" of the BEC descriptions classification table". Need to invest age further and add missing information to classifications table. Another warning message "3769 codes in the "HS.Code_6" column of the BEC descriptions classification table are not in the "HS.Code_6" column of the trades statistics table"- very possible

- For merging principle commodities and BEC have to think through strategy to check missing with UN outline 

- Check number of rows with NAs present (line 218), not sure what this code is doing- is it a follow on from line 104, highlighting PRF NAs?

- Tried to run code (line 235) checking commodity values against expected values and get error "Error in match.names(clabs, names(xi)) : names do not match previous names"

Script 2 Formatted Tables

-  Code is working well. Need to develop a Table 11, and write code for table 14. Additionally quality check all outputs against previous mythology 

Script 3 Build Monthly Report

- Loading packages getting error with "flextable" (line 14). Have used stack overflow link (https://stackoverflow.com/questions/48199737/cant-install-flextable) and install Homebrew but still getting error below, maybe a Mac issue that was highlighted in one of the comments 

"Error: package or namespace load failed for ‘flextable’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so, 6): Library not loaded: /opt/X11/lib/libfreetype.6.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so
  Reason: image not found"

- NSDP Indicators, have taken out ECO 1.4.1- no longer think can be used as a Proxy. The Indicators have are ECO 1.5.2; 1.5.3; 1.6.2; 1.7.1. Also think we could add ENV 1.3.1 and ECO 4.3.1; 4.3.2. However for now will focus on ECO 1.5.2-1.7.1. Below is link to NSDP M&E Framework 
https://www.gov.vu/images/publications/NSDP%20M&E%20Framework.pdf

- Indicator 1.5.3, code from line 122-125, is this needed? Not quite sure what it is doing

- Indicator 1.6.2, used hard coding in line 139, how do I adapt this?

- Calculate Target 1.5.2 (line 170). Using MSG, however in 2019 data PICTA also used, have to add MSG and PICTA to all calculations of Trade Agreements (Line 173). Hard coding again used in line 188- trying to compare current month being analysed to historical months to estimate whether Target being achieved

- Calculate Target 1.5.3 (line 192). Have created baseline datasets, using historical data "imports_HS_summaryStats_02-10-20.csv" and "exports_HS_summaryStats_02-10-20.csv", subset of 2016. Not sure how to calculate monthly Trade balance for major partner countries per month.

- Calculate Target 1.6.2 (line 203). Not sure how to calculate value of trade agreement of GDP per month using 2016 GDP to create baseline to compare current data too. 

- Calculate Target 1.7.1 (line 220). Similar issue to target 1.5.3. Not sure how to calculate monthly major commodities per month.

- Trade Balance by Major Partner Countries: The Data Frame is created, not sure how to visualise it with a map. Most likely need to add a table too to highlight, exports and imports

- Trade Balance of Pacific islands: Hard coding used again lines 396 & 426. Have calculated top 5 imports and exports from Pacific countries. Need to exclude countries with 0, and create bar chart for all others, not sure how to do this

- Trade Balance of New Emerging Markets: Is it possible to talk me through this code, not following it correctly?

- Trade by Trade Agreement: Is it possible to talk me through this code, not following it correctly?

-

**Comment by JosephCrispell** [comment]
# Script 1- Processing

> * Will share the CSV that is immediately downloaded from the Customs system for April. Will delete the identifiable data. Will have to write code to potentially delete these columns and rows. Potentially do this at the end and produce two csv's, one that is anonymised and another with individual information

Is this the `SEC_PROC_ASY_AnnoymisedRawDataAndReferenceTables_30-04-21.csv` file you shared in Skype? We'll also need to the formatted tables up until March 2021 - sorry if you have already sent this and I missed it! 😊

> * Converting SITC to character (line 57), SITC Code 8431.2 isn't in convertible format. Anna checked this in HS Code tables that have corresponding SITC codes, and in fact SITC code should be 0744.92. Have gone back into raw data and changed. Now running no issues

Great! 👍

> * Explore extent of missing data (line 84), checking for large amounts of missing data (>0.1). Can we also look at missing data if only a few cells, maybe create data-frames for each column with missing data 9similar to PRF data-frame)?

[NEED TO CHECK THIS]

> * Extract 3 and 1 digit SITC Codes (line 100 & 101), two null or NA values in SITC column skipped (HS Codes- 71189000 & 99050000). Checked HS Tables, document states HS Code 71189000 corresponding SITC is null (will have to confirm with Customs), but HS Code 99050000 corresponding SITC is 0990.15- have gone back into raw data and changed. Now only one running issue.

Yes, I think these are carried through from `NA` values in the SITC column.

> * NAs present in any of the above columns (line 104), code View(tradeStatsCommodities[is.na(tradeStatsCommodities$column_name), ]) may not be working

[NEED TO CHECK THIS

> * Been through checkingMergingColumnsForClassificationTables() function, would it be possible to quickly explain how this works? Also not sure of the format of the data-frame produced "missingValuesHSCode_2" could explain also?

[NEED TO ADD HERE]

> * Merging SITC with cleaned data (line 158). Warning message 1 code in SITC cleaned data that not in classifications. Again not sure of format of data-frame "missingValuesSITC_1" (this to be similar to other merges)

[NEED TO ADD HERE]

> * Merging Import with cleaned data (line 171). Warning message "185 codes in the "CO" column of the Import country classification table are not in the "CO" column of the trades statistics table"- very possible.

Yes, we can discuss these warning messages - I think it's good to have so we're always checking the classification tables but in many cases missing data are expected because classification tables often represent a subset of the data. 

> * Merging export with cleaned data (line 184). Warning message "194 codes in the "CE.CD" column of the Export country classification table are not in the "CE.CD" column of the trades statistics table"- very possible

As above.

> * Merging principle commodities with cleaned data (line 198). Warning message "378 codes in the "HS.Code" column of the trades statistics table are not in the "HS.Code" of the Principle Commodities classification table". Need to invest age further and add missing information to classifications table. Also how is this possible when merge previously change in observations is only about 5? Another warning message "3759 codes in the "HS.Code" column of the Principle Commodities classification table are not in the "HS.Code" column of the trades statistics table"- very possible

As above.

> * Merging BEC values with cleaned data (line 211). Warning message "611 codes in the "HS.Code_6" column of the trades statistics table are not in the "HS.Code_6" of the BEC descriptions classification table". Need to invest age further and add missing information to classifications table. Another warning message "3769 codes in the "HS.Code_6" column of the BEC descriptions classification table are not in the "HS.Code_6" column of the trades statistics table"- very possible

As above.

> * For merging principle commodities and BEC have to think through strategy to check missing with UN outline
We should revisit this at our next catch up - if there's a UN BEC table available a quick check should be pretty straightforward.

> * Check number of rows with NAs present (line 218), not sure what this code is doing- is it a follow on from line 104, highlighting PRF NAs?

[ADD EXPLANATION HERE]

> * Tried to run code (line 235) checking commodity values against expected values and get error "Error in match.names(clabs, names(xi)) : names do not match previous names"

[NEED TO LOOK INTO THIS]

**Comment by JosephCrispell** [comment]
# Script 2 Formatted Tables

> Code is working well. Need to develop a Table 11, and write code for table 14. Additionally quality check all outputs against previous mythology

Great, and those are good next steps 👍

**Comment by JosephCrispell** [comment]
# Script 3 Build Monthly Report
 
> * Loading packages getting error with "flextable" (line 14). Have used stack overflow link (https://stackoverflow.com/questions/48199737/cant-install-flextable) and install Homebrew but still getting error below, maybe a Mac issue that was highlighted in one of the comments

    "Error: package or namespace load failed for ‘flextable’ in dyn.load(file, DLLpath = DLLpath, ...):
    unable to load shared object 
    '/Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so':
    dlopen(/Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so, 6): Library not 
    loaded: /opt/X11/lib/libfreetype.6.dylib
    Referenced from: /Library/Frameworks/R.framework/Versions/3.6/Resources/library/systemfonts/libs/systemfonts.so
    Reason: image not found"

---

> * NSDP Indicators, have taken out ECO 1.4.1- no longer think can be used as a Proxy. The Indicators have are ECO 1.5.2; 1.5.3; 1.6.2; 1.7.1. Also think we could add ENV 1.3.1 and ECO 4.3.1; 4.3.2. However for now will focus on ECO 1.5.2-1.7.1. Link to NSDP M&E Framework: https://www.gov.vu/images/publications/NSDP%20M&E%20Framework.pdf

---

> * Indicator 1.5.3, code from line 122-125, is this needed? Not quite sure what it is doing

---

> * Indicator 1.6.2, used hard coding in line 139, how do I adapt this?

---

> * Calculate Target 1.5.2 (line 170). Using MSG, however in 2019 data PICTA also used, have to add MSG and PICTA to all calculations of Trade Agreements (Line 173). Hard coding again used in line 188- trying to compare current month being analysed to historical months to estimate whether Target being achieved

---

> * Calculate Target 1.5.3 (line 192). Have created baseline datasets, using historical data "imports_HS_summaryStats_02-10-20.csv" and "exports_HS_summaryStats_02-10-20.csv", subset of 2016. Not sure how to calculate monthly Trade balance for major partner countries per month.

---

> * Calculate Target 1.6.2 (line 203). Not sure how to calculate value of trade agreement of GDP per month using 2016 GDP to create baseline to compare current data too.

---

> * Calculate Target 1.7.1 (line 220). Similar issue to target 1.5.3. Not sure how to calculate monthly major commodities per month.

---

> * Trade Balance by Major Partner Countries: The Data Frame is created, not sure how to visualise it with a map. Most likely need to add a table too to highlight, exports and imports

---

> * Trade Balance of Pacific islands: Hard coding used again lines 396 & 426. Have calculated top 5 imports and exports from Pacific countries. Need to exclude countries with 0, and create bar chart for all others, not sure how to do this

---

> * Trade Balance of New Emerging Markets: Is it possible to talk me through this code, not following it correctly?

---

> * Trade by Trade Agreement: Is it possible to talk me through this code, not following it correctly?

**Comment by hugo-e-pigott** [comment]
*Script 1

Create API to connect to Customs System. This was Stanley’s response who leads on their database. 

<img width="618" alt="Screenshot 2021-08-13 at 07 58 09" src="https://user-images.githubusercontent.com/65749107/129284281-8c80690c-d85d-49ed-af95-1c9e996a9ede.png">

Line 29: “Delete header row from raw data”- this code is not working. When read in data, the top row is blank- not the column headers. This must be deleted and then row below made into headers. Have tried a few different ways to write the code but none working.

<img width="987" alt="Screenshot 2021-08-13 at 07 54 50" src="https://user-images.githubusercontent.com/65749107/129284375-8b46aeb6-e6f2-4155-a4ad-8d9384b94aaf.png">

Line 87: After “Set the plotting margins” get a blank pop up that says “Quartz”. Not sure what this is unless supposed to be the bar chart visualisation of missing data…. But as stated the pop up is blank

Merging
•	Principle Commodities: Currently 378 codes missing from classification table- have to create code to take all rows of this data and extract to CSV, Anna can then review before integrating into the classification table. Also, after merging the number of observations changes from 12,539 to 12,544, why is this? 
•	BEC Classifications: Got classifications from UN in region of 50,000 however only realised after merging that a number of the HS.Code_6 are duplicates. After using new classifications in merge went down from 628 missing codes from classification table to 233. However, number of observations jumped from 12,544 to 175,995! Two questions: How do I delete duplicates from a specific column (HS.Code_6)? Same as above create code to take all rows of data that is not in classification table, create a data-frame and extract to csv

Line 245: “Check the commodity values against expected values based on historic data” this is still giving an error.

<img width="390" alt="Screenshot 2021-08-13 at 08 24 21" src="https://user-images.githubusercontent.com/65749107/129284405-af825553-6a0e-4c26-baea-2201f6f92fe2.png">


*Script 2

Table 11
•	Line 658 “Get the trade stats for the exports aligned to the MSG agreement”. The PRF column records imports aligned to the MSG agreement but not exports. So have merged MSG agreement commodities into dataset, and then used column “Not.Included.in.MSG” to filter out right information. The code here I have written seems extremely long winded, is there way to shorten it? 
•	Line 670 & 686: Checking here value differences between using group_by() and creating data-frame to look at top ten exporting commodities. Here figures are different, wasn’t sure why this is. Maybe because group_by() includes exports and reexports (1000, 3071) and data-frame only exports (1000)?
•	Create table and extract to excel xlsx output sheet 11. To show Top ten exports highlighted in the data-frame (simple table- top ten exports aligned to MSG and their statistical value)
•	Similar issues from imports from previous two points (figures not matching, create table to extract and add to excel)

Table 14
•	Have to sort out classifications, but think code looks good to create data-frame. Just have to format into table and extract into excel

*Script 3 (Overall comment, possible to clean/reduce code to be more efficient) 

NSDP Indicators (Policy, Indicator, Target)

<img width="755" alt="Screenshot 2021-08-13 at 10 14 33" src="https://user-images.githubusercontent.com/65749107/129284459-b7e263c6-9339-4a21-85fc-1f5ff4bc0e90.png">

•	NSDP Indicators 1.5.2, 1.5.3, 1.6.2, 1.71 seem ok. Would be possible to review code and clean up? 
•	NSDP Indicator Targets: there is hard coding when choosing current month need to get rid of. Target 1.7.1 not properly working (merge not functioning properly maybe)
•	Still have to adapt table and connect to indicators & targets 

Trade Balance by Major Partner Countries
•	Need to take data-frame and visualise it on the map. If space maybe worth adding small table next to map, is this possible with specific values for countries?

Trade Balance of Pacific Countries
•	Exclude those with zero exports and imports 

Trade Balance of new emerging markets
•	Need to add values above or below bars, as current difficult to see exports as values are small compared to imports

Trade by Trade Agreement (statistical value)
•	Create cross tab table of exports top 5 to specific pacific islands part of MSG (rows to have exports, column headers country- Fiji, New Caledonia, Papa New Guinea, Solomon Islands)
•	Create cross tab table of imports top 5 from specific pacific islands part of MSG (rows to have imports, column headers country- Fiji, New Caledonia, Papa New Guinea, Solomon Islands)

Principle Exports & Imports
•	Have to take current monthly calculation “majorExportsCurrentMonth” &  “majorImportsCurrentMonth”and add to historical data-frame. Then visualise monthly values

Top 5 new Major Exports and Imports
•	“Select top 5 new exports” & “Select top 5 new imports”. Tried to select top five rows using this piping function but not working. 

Imports of Products related to Health
•	Still have to do

**Comment by hugo-e-pigott** [comment]
Also as a reminder it looks as though some products such as Kava roots are not coded under kava and I found them when searching data frame processedTradeStats and searching "Goods.Comm..Dsc" column.

Is it possible to search for words in R?

**Comment by JosephCrispell** [comment]
> Create API to connect to Customs System. This was Stanley’s response who leads on their database.

We can chat about this but I think the next step is to start building a url string and using the functions in the [httr](https://cran.r-project.org/web/packages/httr/httr.pdf) and [jsonlite](https://cran.r-project.org/web/packages/jsonlite/jsonlite.pdf) packages to run some queries. Something like what is being done in [page 2 on this tutorial](https://www.programmableweb.com/news/how-to-access-any-restful-api-using-r-language/how-to/2017/07/21?page=2).

**Comment by JosephCrispell** [comment]
> Line 29: “Delete header row from raw data”- this code is not working. When read in data, the top row is blank- not the column headers. This must be deleted and then row below made into headers. Have tried a few different ways to write the code but none working.
> 
> <img alt="Screenshot 2021-08-13 at 07 54 50" width="987" src="https://user-images.githubusercontent.com/65749107/129284375-8b46aeb6-e6f2-4155-a4ad-8d9384b94aaf.png">

Best thing to do in this case is to use the `skip` parameter in `read.csv()` function. In this case setting `skip = 1` when calling `read.csv()` will mean R starts reading the dataframe from the second row and skips the first line.

**Comment by JosephCrispell** [comment]
> Line 87: After “Set the plotting margins” get a blank pop up that says “Quartz”. Not sure what this is unless supposed to be the bar chart visualisation of missing data…. But as stated the pop up is blank

Weird one, not sure why "Quartz" would be popping up. `par(mar=c(10, 4, 4, 1))` is just setting the plotting margins and I think the plot has now been removed so this can be to.

- [ ] TODO - check with @hugo-e-pigott to replicate this one

**Comment by JosephCrispell** [comment]
> Principle Commodities: Currently 378 codes missing from classification table- have to create code to take all rows of this data and extract to CSV, Anna can then review before integrating into the classification table. Also, after merging the number of observations changes from 12,539 to 12,544, why is this?

Added a TODO here for getting missing data and quality issues flagged in csv for Anna. On the increasing number of rows, this is caused by duplicate values in the `HS.Code` column of the principle commodities table.

- [ ] TODO rethink flagging of missing data and quality issues - map to raw data and export into CSV
- [ ] TODO only merge necessary columns - some duplicates being brought in on merging?

**Comment by JosephCrispell** [comment]
> • BEC Classifications: Got classifications from UN in region of 50,000 however only realised after merging that a number of the HS.Code_6 are duplicates. After using new classifications in merge went down from 628 missing codes from classification table to 233. However, number of observations jumped from 12,544 to 175,995! Two questions: How do I delete duplicates from a specific column (HS.Code_6)? Same as above create code to take all rows of data that is not in classification table, create a data-frame and extract to csv

Yes the observations jumping is worrying - we should add a check for duplicates on the columns used for merging each of the classifications tables. For removing duplicates you can use the [`duplicated()`](https://www.datanovia.com/en/lessons/identify-and-remove-duplicate-data-in-r/) or [`distinct()`](https://www.datasciencemadesimple.com/remove-duplicate-rows-r-using-dplyr-distinct-function/#:~:text=DataScience%20Made%20Simple-,Remove%20Duplicate%20rows%20in%20R%20using%20Dplyr%20%E2%80%93%20distinct%20()%20function,variable%20or%20with%20multiple%20variable.) functions.

- [ ] TODO Add in approach to check for duplicate values in classification ID columns used for merging - remove where present

**Comment by JosephCrispell** [comment]
> Line 245: “Check the commodity values against expected values based on historic data” this is still giving an error.

This will be a column naming issue I think - I think we are working on different versions of the trade statistics data.

- [ ] TODO @hugo-e-pigott to share version of trade statistics data so @JosephCrispell can replicate error

**Comment by JosephCrispell** [comment]
> • Line 658 “Get the trade stats for the exports aligned to the MSG agreement”. The PRF column records imports aligned to the MSG agreement but not exports. So have merged MSG agreement commodities into dataset, and then used column “Not.Included.in.MSG” to filter out right information. The code here I have written seems extremely long winded, is there way to shorten it?

A slightly simpler approach that avoids merging:
```
msgAgreementCommoditiesExcludeFile <- file.path(openDataFolder, "OPN_FINAL_ASY_MSGClassifications_31-01-20.csv")
msgAgreementCommoditiesExclude <- read.csv(msgAgreementCommoditiesExcludeFile, header=TRUE, na.strings=c("","NA", "NULL", "null"))
hsCodesOfCommoditiesToExclude <- msgAgreementCommoditiesExclude[msgAgreementCommoditiesExclude$Not.Included.in.MSG == "Yes", "HS.Code_6"]
hsCodesOfCommoditiesToExclude <- sapply(hsCodesOfCommoditiesToExclude, FUN=padWithZeros, "HS", 6)
tradeStatsForMSGExports <- processedTradeStats[processedTradeStats$HS.Code_6 %in% hsCodesOfCommoditiesToExclude == FALSE, ]
```

**Comment by JosephCrispell** [comment]
> • Line 670 & 686: Checking here value differences between using group_by() and creating data-frame to look at top ten exporting commodities. Here figures are different, wasn’t sure why this is. Maybe because group_by() includes exports and reexports (1000, 3071) and data-frame only exports (1000)?
> • Create table and extract to excel xlsx output sheet 11. To show Top ten exports highlighted in the data-frame (simple table- top ten exports aligned to MSG and their statistical value)
> • Similar issues from imports from previous two points (figures not matching, create table to extract and add to excel)

Not sure I quite understand these - added a TODO below to discuss tomorrow.

- [ ] TODO Discuss these tasks to add clarity

**Comment by JosephCrispell** [comment]
> Table 14
> • Have to sort out classifications, but think code looks good to create data-frame. Just have to format into table and extract into excel

- TODO Add code to insert formatted table into excel file

**Comment by JosephCrispell** [comment]
> NSDP Indicators 1.5.2, 1.5.3, 1.6.2, 1.71 seem ok. Would be possible to review code and clean up?
> • NSDP Indicator Targets: there is hard coding when choosing current month need to get rid of. Target 1.7.1 not properly working (merge not functioning properly maybe)
> • Still have to adapt table and connect to indicators & targets

Probably worth doing a quick run through of these tomorrow so I know where to look. Definitely scope to clean up code in all scripts 👍.

- [ ] TODO Run through code for above indicators to discuss improvements

**Comment by JosephCrispell** [comment]
> Trade Balance by Major Partner Countries
> • Need to take data-frame and visualise it on the map. If space maybe worth adding small table next to map, is this possible with specific values for countries?

Should be possible but the table would end up being quite small so I'm not sure if it's worth it. First step is to get values reflected on map and add legend.

- [ ] TODO Update map with true values and add legend

**Comment by JosephCrispell** [comment]
> Trade Balance of new emerging markets
> • Need to add values above or below bars, as current difficult to see exports as values are small compared to imports

Yes definitely possible and I agree would be helpful. Here's quite a nice answer on StackOverflow about [how to add labels above bars with ggplot](https://stackoverflow.com/questions/6455088/how-to-put-labels-over-geom-bar-in-r-with-ggplot2)

**Comment by JosephCrispell** [comment]
> Also as a reminder it looks as though some products such as Kava roots are not coded under kava and I found them when searching data frame processedTradeStats and searching "Goods.Comm..Dsc" column.
> 
> Is it possible to search for words in R?

Yes definitely R has a suite of functions for searching for patterns in text. You could take a look at the [`grep()`](https://www.nbshare.io/notebook/429754141/How-To-Use-Grep-In-R/) and [`str_detect`](https://stringr.tidyverse.org/reference/str_detect.html) functions.

- TODO @hugo-e-pigott to add a little more detail here so @JosephCrispell can point to solution using pattern based search.

**Comment by JosephCrispell** [comment]
Hi @hugo-e-pigott,

As you'll see above I've added some comments to some of the tasks and questions you've noted. I've also added TODO checkboxes where there's a particular thing for the both of us to consider.

One general note is that this issue has become a little unwieldy and lengthy - it might be worthwhile splitting the tasks across multiple issues so it's a little clearer to navigate through the conversations. 

I hope the above comments are helpful and looking forward to discussing tomorrow.

Joe
